### PR TITLE
PDDLAnytimePlanner refactoring

### PR DESCRIPTION
**Pull request by Framba-Luca**

### CODE REVIEW

File: unified_planning/test/pddl/enhsp.py [hunk 1]
```diff
@@ -64,81 +91,14 @@ def _result_status(
...
-            "hadd",
-            "-s",
-            "gbfs",
-            "-anytime",
-        ]
+    def _ending_plan_str(self) -> str:
+        return "Plan-Length:"
 
-        if timeout is not None:
-            self._options.extend(["-timeout", str(timeout)])
```
Comment (alvalentini) [inline unified_planning/test/pddl/enhsp.py:87]: This part should be kept

File: unified_planning/test/pddl/enhsp.py [hunk 2]
```diff
@@ -14,19 +14,24 @@
...
 
 FILE_PATH = os.path.dirname(os.path.abspath(__file__))
 
 
-class ENHSP(up.engines.PDDLPlanner, up.engines.mixins.AnytimePlannerMixin):
+class ENHSP(PDDLAnytimePlanner):
     def __init__(self):
-        up.engines.PDDLPlanner.__init__(self, False, True)
+        PDDLAnytimePlanner.__init__(self, False, True)
         self._options = ["-planner", "opt-hrmax"]
```
Comment (alvalentini) [inline unified_planning/test/pddl/enhsp.py:35]: There is no need to keep this two options here now

File: unified_planning/engines/pddl_anytime_planner.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        """
+        Returns the string representing the ending of a plan in the engine's output.
+        """
+        raise NotImplementedError
+
+    def _parse_plan_line(self, plan_line: str) -> str:
+        """
+        Takes an engine's output line in between the starting_plan string and the engine_plan
+        string and returns the string representing the parsed plan.
+        TODO document this better
```
Comment (alvalentini) [inline unified_planning/engines/pddl_anytime_planner.py:167]: TODO

**Review by alvalentini — CHANGES_REQUESTED** [review]

**Comment by alvalentini** [comment]
@roeger @hstairs @IvanSerina 

This PR aims to implement a common PDDLAnytimePlanner to avoid code duplication between planning engines.

Any feedback?

File: unified_planning/engines/pddl_anytime_planner.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    def _parse_plan_line(self, plan_line: str) -> str:
+        """
+        Takes an engine's output line in between the starting_plan string and the engine_plan
+        string and returns the string representing the parsed plan.
+        The representation to obtain is:
+        ``(action-name (param1) (param2) (param3...))`` in each line for SequentialPlans
+        ``(start-time) (action-name (param1) (param2) (param3...)) (duration)`` in each line for TimeTriggeredPlans,
+        where ``(duration)`` is optional
+        """
+        raise NotImplementedError
```
Comment (Framba-Luca) [inline unified_planning/engines/pddl_anytime_planner.py:172]: To fix doc

**Review by alvalentini — APPROVED** [review]

**Comment by roeger** [comment]
@alvalentini I just tried to use the new class in the Fast Downward integration but there is a situation, where I am not sure how to cover it. The anytime search in Fast Downward terminates without a timeout once the search space has been exhausted. If a plan has been found earlier, my integration previously returned the last found plan with status SOLVED_SATISFICING.

With the current implementation, I would have a sequence of intermediate plans that at some point stops without any final plan. Is this the user interface we want? If not, what's your advice?

**Pull request merged**